spec/entities/base_spec.rb

```diff
@@ -51,37 +59,103 @@
   end
 
   describe 'request' do
-    let(:fixture) { File.read("spec/savon/fixtures/project/does_not_exists.xml") }
-    let(:client) { mock('client') }
-    let(:response) { mock('response') }
-    let(:request) { mock('request') }
-    let(:options) { mock('options') }
+    context 'with successful responses' do
+      let(:exists_response) { {'exists' => false} }
+      let(:bases_response) { {'bases' => {'id' => 1}} }
+      let(:prefix_bases_response) { {'bases' => {'id' => 2}} }
+
+      let(:connection) { Faraday.new { |builder| builder.adapter :test do |stubs|
+          stubs.get('/bases/1/exists') { |env| [200, {}, exists_response] }
+          stubs.get('/bases/') { |env| [200, {}, bases_response] }
+          stubs.get('/prefix/bases/') { |env| [200, {}, prefix_bases_response] }
+        end
+      }}
 
-    before :each do
-      options.expects(:timeout=)
-      options.expects(:open_timeout=)
-      request.expects(:url).with('/bases/exists')
-      request.expects(:body=).with({id: 1})
-      request.expects(:options).twice.returns(options)
+      before :each do
+        subject.class.stubs(:client).returns(connection)
```

Comment:
Sorry here is the exact example on why this is necessary: https://github.com/mezuro/kalibro_client/pull/78#discussion-diff-40673185R322
It's fact that even on the branch wihin weeks there is already a unused stub.

If someday we get good at TDD we may talk about replacing mocks by stubs. While this day do not come, please stick to mocks since we can not afford this kind of maintenance cost of outdated specs.